src/charts/c3/c3-chart.component.js

```diff
@@ -97,9 +97,9 @@
           chartData = ctrl.config;
           if (chartData) {
             chartData.bindto = '#' + $attrs.id;
-            // always re-generate donut pct chart because it's colors
-            // change based on data and thresholds
-            if (!chart || $attrs.id.indexOf('donutPctChart')) {
+            // only re-generate donut pct chart if it has a threshold object
+            // because it's colors will change based on data and thresholds
+            if (!chart || ($attrs.id.includes('donutPctChart') && chartData.thresholds)) {
```

Comment:
Sure, I have reversed the usage of 'includes', and will instead will use the indexOf() !== -1.

For reference, is there a particular reason why indexOf() would be more advantageous than includes()?